music21/musicxml/m21ToXml.py

```diff
@@ -3472,7 +3521,23 @@ def noteToXml(self, n: note.GeneralNote, noteIndexInChord=0, chordParent=None):
             for t in mxTieList:
                 mxNote.append(t)
 
-        # TODO: instrument
+        # instrument tags are necessary when there is more than one
+        if self.parent is not None and len(self.parent.instrumentStream) > 1 and not n.isRest:
+            if chordParent:
+                closest_inst = chordParent.getContextByClass(Instrument, followDerivation=False)
+                assert closest_inst in self.parent.instrumentStream
+            else:
+                closest_inst = n.getContextByClass(Instrument, followDerivation=False)
+                assert closest_inst in self.parent.instrumentStream
+            instance_to_use = self.parent.instrumentStream[type(closest_inst)].first()
```

Comment:
As before -- if the first instance of each Instrument were cached, this would be both faster and cleaner:

```
instance_id_to_use = self.getScoreInstrumentId(closest_inst)
```

or something like that.